evm_loader/lib/src/syscall_stubs.rs

```diff
@@ -22,43 +36,130 @@ impl EmulatorStubs {
             .unwrap_or_default();
         let rent = bincode::deserialize(&data).map_err(|_| ProgramError::InvalidArgument)?;
 
-        Ok(Box::new(Self { rent }))
+        Ok(Box::new(Self {
+            rent,
+            original_stubs,
+        }))
     }
 }
 
 impl SyscallStubs for EmulatorStubs {
     fn sol_get_rent_sysvar(&self, pointer: *mut u8) -> u64 {
-        unsafe {
-            #[allow(clippy::cast_ptr_alignment)]
-            let rent = pointer.cast::<Rent>();
-            *rent = self.rent;
+        if is_original_stubs_for_thread() {
+            self.original_stubs.sol_get_rent_sysvar(pointer)
+        } else {
+            unsafe {
+                #[allow(clippy::cast_ptr_alignment)]
+                let rent = pointer.cast::<Rent>();
+                *rent = self.rent;
+            }
+            0
         }
-
-        0
     }
```

Comment:
No, we can't. The main problem is that `SyscallStubs` is global for the whole program. Moreover, it can be called in different contexts: our emulation and SolanaProgramTest bank emulator. We have different threads for API utility, and each of these can be in context. So our `SyscallStubs` implementation provides implementation for our emulation purposes, but for specific thread it can be switched to original `SyscallStubs` implementaion.

Reply:
Ah in that case, I think I should try to do what I am suggesting myself and seeing where I get stuck, or not.